src/main/java/seedu/txtdata/TxtFileStatus.java

```diff
@@ -80,47 +82,96 @@ public static void saveExpenditureList(ArrayList<Expenditure> expenditures) thro
-            }
-            switch (saveData[INDEX_TYPE]) {
-            case "Acad":
-                initializeAcademicExpenditure(saveData, expenditures);
-                break;
-            case "Accom":
-                initializeAccommodationExpenditure(saveData, expenditures);
-                break;
-            case "B":
-                initializeBorrowExpenditure(saveData, expenditures);
-                break;
-            case "En":
-                initializeEntertainmentExpenditure(saveData, expenditures);
-                break;
-            case "F":
-                initializeFoodExpenditure(saveData, expenditures);
-                break;
-            case "L":
-                initializeLendExpenditure(saveData, expenditures);
-                break;
-            case "O":
-                initializeOtherExpenditure(saveData, expenditures);
-                break;
-            case "Tr":
-                initializeTransportExpenditure(saveData, expenditures);
-                break;
-            case "Tu":
-                initializeTuitionExpenditure(saveData, expenditures);
-                break;
-            default:
-                throw new ArrayIndexOutOfBoundsException();
+            try {
+                String saveString = s.nextLine();
+                String[] saveData = saveString.split("d/|v/|t/|p/|n/|o/|r/");
+                checkNegative(saveData[INDEX_VALUE]);
+                switch (saveData[INDEX_TYPE]) {
+                case "Acad":
+                    initializeAcademicExpenditure(saveData, expenditures);
+                    break;
+                case "Accom":
+                    initializeAccommodationExpenditure(saveData, expenditures);
+                    break;
+                case "B":
+                    initializeBorrowExpenditure(saveData, expenditures);
+                    break;
+                case "En":
+                    initializeEntertainmentExpenditure(saveData, expenditures);
+                    break;
+                case "F":
+                    initializeFoodExpenditure(saveData, expenditures);
+                    break;
+                case "L":
+                    initializeLendExpenditure(saveData, expenditures);
+                    break;
+                case "O":
+                    initializeOtherExpenditure(saveData, expenditures);
+                    break;
+                case "Tr":
+                    initializeTransportExpenditure(saveData, expenditures);
+                    break;
+                case "Tu":
+                    initializeTuitionExpenditure(saveData, expenditures);
+                    break;
+                default:
+                    throw new TxtFileException();
+                }
+            } catch (TxtFileException t) {
+                System.out.println(t.getMessage());
+            } catch (DateTimeParseException | NumberFormatException | ArrayIndexOutOfBoundsException e) {
```

Comment:
Good working catching the exception in txtFileStatus instead for a more compact MyLedger.